maven-parent/pom.xml

```diff
@@ -33,20 +33,25 @@
     <ozone.version>1.0.0-SNAPSHOT</ozone.version>
   </properties>
 
-  <dependencies>
-    <dependency>
-      <groupId>com.ozonehis</groupId>
-      <artifactId>ozone</artifactId>
-      <type>zip</type>
-      <version>${ozone.version}</version>
-    </dependency>
-    <dependency>
-      <groupId>com.ozonehis</groupId>
-      <artifactId>ozone-scripts</artifactId>
-      <type>zip</type>
-      <version>${ozone.version}</version>
-    </dependency>
-  </dependencies>
+  <profiles>
+    <profile>
+      <id>parentBuild</id>
+      <dependencies>
+        <dependency>
+          <groupId>com.ozonehis</groupId>
+          <artifactId>ozone</artifactId>
+          <type>zip</type>
+          <version>1.0.0-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+          <groupId>com.ozonehis</groupId>
+          <artifactId>ozone-scripts</artifactId>
+          <type>zip</type>
+          <version>1.0.0-SNAPSHOT</version>
+        </dependency>
+      </dependencies>
+    </profile>
+  </profiles>
```

Comment:
@ibacher , the Maven Parent build needs to go through, so if I just remove the dependencies, the build will fail. I have made them running behind a profile, so that child projects won't be bothered by them.

This forces to build Ozone with `-P parentBuild` though.
Do you know if there is a way around this? Eg, profile activation controlled in the root pom.

Reply:
Maven profiles are horribly inflexible. The one thing I can think of is it possible to default-activate a profile based on an environment variable and most CI servers set the CI environment variable to `true`, so you could use:

```xml
<activation>
  <property>
    <name>env.CI</name>
     <value>true</value>
  </property>
</activation>
```

To turn it on by default in CI.

This change, however, is kind of a regression: the advantage of having the Ozone version specified in the parent is that upgrading Ozone is just a matter of switching the parent version (assuming released versions) and, if you need a specific version, the child project should always be able to override the `${ozone.version}` property to use a different Ozone version. (The actual version of Ozone downloaded is whatever the local Maven configuration specifies, but maybe I'm missing something).

Reply:
>the child project should always be able to override the ${ozone.version} property to use a different Ozone version

What I want the lower-level project to do is to re-use the parent (all complicated build steps that it brings) but not depend on Ozone at all and instead on the higher level distro (Ozone Haiti in this example).
Ozone Haiti being an Ozone distribution itself, all the same build steps as a base Ozone can be applied.

Reply:
Would it make sense, then, to treat Ozone Haiti as an Ozone distribution and just unpack a version of that in Ozone HSC? The main thing you get from maven-parent if you don't want all the build stuff is just maven-commons, so Ozone HSC could inherit from that.